extends Control

@onready var stats = $Panel/Msg/MsgLabel
@onready var morale = $Panel/Morale
@onready var total_enemies = get_parent().get_node("PathSpawner").total_enemies
var enemies_killed = 0
var enemies_passed = 0

# Called when the node enters the scene tree for the first time.
func _ready() -> void:
	pass

# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(_delta: float) -> void:
	enemies_passed = get_parent().get_node("LevelContainer").get_child(0).get_node("EndZone").enemy_count
	enemies_killed = total_enemies - enemies_passed
	display_end_phrases()
	pass

var morale_phrases = [
	"Chaque annÃ©e, environ 8 millions de tonnes de plastique finissent dans nos ocÃ©ans.",
	"Environ 100 millions d'animaux marins meurent chaque annÃ©e Ã  cause de la pollution.",
	"Environ 50 % des rÃ©cifs coralliens dans le monde ont Ã©tÃ© perdus au cours des 30 derniÃ¨res annÃ©es.",
	"Plus de 1 million d'espÃ¨ces marines sont actuellement menacÃ©es d'extinction.",
	"Il existe plus de 400 zones mortes dans les ocÃ©ans, des rÃ©gions oÃ¹ le niveau d'oxygÃ¨ne est trop bas pour soutenir la vie marine.",
	"La biodiversitÃ© marine a diminuÃ© de 39 % en moyenne depuis 1970."
]

func display_end_phrases() -> void:
	stats.bbcode_text = "[center]En ramassant " + str(enemies_killed) + " dÃ©chets, vous avez protÃ©gÃ© l'ocÃ©an de la pollution.\n\nVotre dÃ©termination a fait la diffÃ©rence ! ðŸŒŠ\n\nCependant, restez conscients que " + str(enemies_passed) + " dÃ©chets sont tout de mÃªme passÃ©s pour se jeter dans notre bel ocÃ©an.\n\nNe lÃ¢chez pas la lutte ![/center]"
	
func display_moral() -> void:
	var shuffled_phrases = morale_phrases.duplicate()
	shuffled_phrases.shuffle()
	var selected_phrase = shuffled_phrases[0]
	morale.bbcode_text = "[center][b][i]" + selected_phrase + "[/i][/b][/center]"

func _on_rejouer_pressed() -> void:
	var menu_scene = preload("res://Scene/menu.tscn")
	get_tree().change_scene_to_packed(menu_scene)
